// @clerk/tanstack-react-start@1.1.3/dist/client/ClerkProvider.js
import { InternalClerkProvider } from '@clerk/react/internal';
import { ScriptOnce } from '@tanstack/react-router';
import { getGlobalStartContext } from '@tanstack/react-start';
import { useEffect } from 'react';
import { isClient } from '../utils/index.js';
import { ClerkOptionsProvider } from './OptionsContext.js';
import { useAwaitableNavigate } from './useAwaitableNavigate.js';
import { pickFromClerkInitState, mergeWithPublicEnvs, parseUrlForNavigation } from './utils.js';
export * from '@clerk/react';
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';

// src/client/ClerkProvider.tsx
var SDK_METADATA = {
  name: "@clerk/tanstack-react-start",
  version: "1.1.3"
};
var awaitableNavigateRef = { current: void 0 };
function ClerkProvider({
  children,
  ...providerProps
}) {
  const awaitableNavigate = useAwaitableNavigate();
  const clerkInitialState = getGlobalStartContext()?.clerkInitialState ?? {};
  useEffect(() => {
    awaitableNavigateRef.current = awaitableNavigate;
  }, [awaitableNavigate]);
  const clerkInitState = isClient() ? window.__clerk_init_state : clerkInitialState;
  const { clerkSsrState, __keylessClaimUrl, __keylessApiKeysUrl, ...restInitState } = pickFromClerkInitState(
    clerkInitState?.__internal_clerk_state
  );
  const mergedProps = {
    ...mergeWithPublicEnvs(restInitState),
    ...providerProps
  };
  const keylessProps = __keylessClaimUrl ? {
    __internal_keyless_claimKeylessApplicationUrl: __keylessClaimUrl,
    __internal_keyless_copyInstanceKeysUrl: __keylessApiKeysUrl
  } : {};
  return /* @__PURE__ */ jsxs(Fragment, { children: [
    /* @__PURE__ */ jsx(ScriptOnce, { children: `window.__clerk_init_state = ${JSON.stringify(clerkInitialState)};` }),
    /* @__PURE__ */ jsx(ClerkOptionsProvider, { options: mergedProps, children: /* @__PURE__ */ jsx(
      InternalClerkProvider,
      {
        initialState: clerkSsrState,
        sdkMetadata: SDK_METADATA,
        routerPush: (to) => {
          const { search, hash, ...rest } = parseUrlForNavigation(to, window.location.origin);
          return awaitableNavigateRef.current?.({
            ...rest,
            search,
            hash,
            replace: false
          });
        },
        routerReplace: (to) => {
          const { search, hash, ...rest } = parseUrlForNavigation(to, window.location.origin);
          return awaitableNavigateRef.current?.({
            ...rest,
            search,
            hash,
            replace: true
          });
        },
        ...mergedProps,
        ...keylessProps,
        children
      }
    ) })
  ] });
}
ClerkProvider.displayName = "ClerkProvider";

export { ClerkProvider };
