// @clerk/tanstack-react-start@1.1.3/dist/server/clerkMiddleware.js
import { createClerkRequest, constants, AuthStatus } from '@clerk/backend/internal';
import { handleNetlifyCacheInDevInstance } from '@clerk/shared/netlifyCacheHandler';
import { createMiddleware } from '@tanstack/react-start';
import { canUseKeyless } from '../utils/feature-flags.js';
import { clerkClient } from './clerkClient.js';
import { resolveKeysWithKeylessFallback } from './keyless/utils.js';
import { loadOptions } from './loadOptions.js';
import { patchRequest, getResponseClerkState } from './utils/index.js';

// src/server/clerkMiddleware.ts
var clerkMiddleware = (options) => {
  return createMiddleware().server(async ({ request, next }) => {
    const clerkRequest = createClerkRequest(patchRequest(request));
    const resolvedOptions = typeof options === "function" ? await options({ url: clerkRequest.clerkUrl }) : options;
    const loadedOptions = loadOptions(clerkRequest, {
      ...resolvedOptions,
      publishableKey: resolvedOptions?.publishableKey,
      secretKey: resolvedOptions?.secretKey
    });
    const {
      publishableKey,
      secretKey,
      claimUrl: keylessClaimUrl,
      apiKeysUrl: keylessApiKeysUrl
    } = await resolveKeysWithKeylessFallback(loadedOptions.publishableKey, loadedOptions.secretKey);
    if (publishableKey) {
      loadedOptions.publishableKey = publishableKey;
    }
    if (secretKey) {
      loadedOptions.secretKey = secretKey;
    }
    const requestState = await clerkClient().authenticateRequest(clerkRequest, {
      ...loadedOptions,
      acceptsToken: "any"
    });
    const locationHeader = requestState.headers.get(constants.Headers.Location);
    if (locationHeader) {
      handleNetlifyCacheInDevInstance({
        locationHeader,
        requestStateHeaders: requestState.headers,
        publishableKey: requestState.publishableKey
      });
      throw new Response(null, { status: 307, headers: requestState.headers });
    }
    if (requestState.status === AuthStatus.Handshake) {
      throw new Error("Clerk: handshake status without redirect");
    }
    const clerkInitialState = getResponseClerkState(requestState, loadedOptions);
    if (canUseKeyless && keylessClaimUrl) {
      clerkInitialState.__internal_clerk_state = {
        ...clerkInitialState.__internal_clerk_state,
        __keylessClaimUrl: keylessClaimUrl,
        __keylessApiKeysUrl: keylessApiKeysUrl
      };
    }
    const result = await next({
      context: {
        clerkInitialState,
        auth: (opts) => requestState.toAuth(opts)
      }
    });
    if (requestState.headers) {
      requestState.headers.forEach((value, key) => {
        result.response.headers.append(key, value);
      });
    }
    return result;
  });
};

export { clerkMiddleware };
