// @clerk/tanstack-react-start@1.1.3/dist/server/keyless/index.js
import { createKeylessService } from '@clerk/shared/keyless';
import { clerkClient } from '../clerkClient.js';
import { createFileStorage } from './fileStorage.js';

// src/server/keyless/index.ts
var keylessServiceInstance = null;
function keyless() {
  if (!keylessServiceInstance) {
    keylessServiceInstance = createKeylessService({
      storage: createFileStorage(),
      api: {
        async createAccountlessApplication(requestHeaders) {
          try {
            return await clerkClient().__experimental_accountlessApplications.createAccountlessApplication({
              requestHeaders
            });
          } catch {
            return null;
          }
        },
        async completeOnboarding(requestHeaders) {
          try {
            return await clerkClient().__experimental_accountlessApplications.completeAccountlessApplicationOnboarding({
              requestHeaders
            });
          } catch {
            return null;
          }
        }
      },
      framework: "tanstack-react-start"
    });
  }
  return keylessServiceInstance;
}

export { keyless };
